common/minimodel/__init__.py

```diff
@@ -1008,3 +1014,56 @@ def total_used(self) -> timedelta:
 
     def observations(self) -> List[Observation]:
         return self.root_group.observations()
+
+
+class Plan:
```

Comment:
Just for discussion:

My thought was that we would have simple types for the plan and the optimizer would handle the timing and assembly of the data. Something like this:

```
@dataclass
class PlanEntry:
    start_time: datetime  # Unsure if this or something else
    obs_id: ObservationID
    atom_start_idx: int
    atom_end_idx: int

NightPlan = List[PlanEntry]
SitePlans = Mapping[NightIndex, NightPlan]
Plans = Mapping[Site, SitePlans]
```

Something like that.

Reply:
(That was just my thinking... not saying we have to go in that direction.)